### DISCUSSION TITLE
魏老师，我是mac系统，最后验证的时候发现没有安装到conda下，不知道哪里错了

### DISCUSSION THREAD

**Discussion by liyingyuan** [weihuayi/fealpy#46] [Installation] [answered]
最后安装的时候：
Using /Library/Python/2.7/site-packages/tornado-6.0.4-py2.7-macosx-10.15-x86_64.egg
Searching for python-dateutil==1.5
Best match: python-dateutil 1.5
python-dateutil 1.5 is already the active version in easy-install.pth

Using **/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python**
Finished processing dependencies for fealpy==1.0
liyingyuan@liyingyuandeMacBook-Pro fealpy % ipython
zsh:**command not found: ipython**
我把anaconda安到了user里，最后cd到fealpy后，安装发现连到的是自带的py2.7;我在终端试图删除py2.7重启，发现也没有用；然后我又试了试
sudo python3 setup_mac.py develop
结果又说没有这个目录
liyingyuan@liyingyuandeMacBook-Pro fealpy % sudo python3 setup_mac.py develop
running develop
**error: can't create or remove files in install directory**

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 2] No such file or directory: '/Library/Python/3.7/site-packages/test-easy-install-7843.write-test'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    /Library/Python/3.7/site-packages/

This directory does not currently exist.  Please create it and try again, or
choose a different installation directory (using the -d or --install-dir
option).
请问老师，这个该怎么解决啊

**Comment by yoczhang** [comment]
有可能你 mac 没安装 python3.7?  `/Library/Python/3.7/site-packages/` 是魏老师指定的安装路径.
你命令中 `sudo python3 setup_mac.py develop` 中应该是 Anaconda 中的 python3,
你可以在 Terminal 中 `cd /Library/Python`, 然后 `ls` 查看是否有 `3.7` 这个目录, 
没有的话, 试着安装一下 python3.7, 然后再试一次.

**Comment by weihuayi** [comment]
我知道原因了， 你把 /Library/Python/3.7/site-packages/ 这个路径手动创建出来
然后再安装就可以了
cd /Library/Python
看一下有没有 3.7 这个目录， 没有就 sudo mkdir 3.7
cd 3.7
sudo mkdir site-packages

**Comment by weihuayi** [comment]
应该不需要再安装 Python 3.7

**Comment by weihuayi** [comment]
你的系统在装 Anaconda 之前已经装了 Python 3.7？

**Comment by liyingyuan** [comment]
一两个月之前装过，但删了，然后刚才看了第一封回复，我又重新装了个python3.7,然后重新装fealpy，中间过程没有错,但是最后验证还是不存在ipython。然后我刚才又把另外装的python3.7卸了，又重新装，结果现在出现了新的问题


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Huayi Wei"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月7日(星期二) 晚上7:59
收件人:&nbsp;"weihuayi/fealpy"<fealpy@noreply.github.com&gt;;
抄送:&nbsp;"天官赐福"<1563025275@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [weihuayi/fealpy] 魏老师，我是mac系统，最后验证的时候发现没有安装到conda下，不知道哪里错了 (#18)





 
你的系统在装 Anaconda 之前已经装了 Python 3.7？

**Comment by yoczhang** [comment]
不用安装 python3.7, 系统只有 2.7.
不过, 既然通过 Anaconda 安装 fealpy, 创建一个 virtual env 是不是更好点,
如果一定要装在 Anaconda 的 base(root) 下的话, 把 fealpy 放在 `anaconda3/lib/python3.7/site-packages/` 是不是也会更好点?

**Comment by weihuayi** [comment]
@yoczhang virtual env 是一个好的手段， 但考虑到大部分人对 Python 还不是很熟悉， 安装过程要尽量的简单，不涉及太多的概念和技巧。 他的这种情况，有点装乱了。

**Comment by liyingyuan** [comment]
我后来也试了 virtual env，但还是不行




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Huayi Wei"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月7日(星期二) 晚上8:18
收件人:&nbsp;"weihuayi/fealpy"<fealpy@noreply.github.com&gt;;
抄送:&nbsp;"天官赐福"<1563025275@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [weihuayi/fealpy] 魏老师，我是mac系统，最后验证的时候发现没有安装到conda下，不知道哪里错了 (#18)





 
@yoczhang virtual env 是一个好的手段， 但考虑到大部分人对 Python 还不是很熟悉， 安装过程要尽量的简单，不涉及太多的概念和技巧。 他的这种情况，有点装乱了。

**Comment by liyingyuan** [comment]
老师，那我要把所有东西删了从头再来一遍吗，我感觉我现在有点乱，就是不知道没删干净之前下的那些东西的话会不会对新的安装有影响



------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Huayi Wei"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月7日(星期二) 晚上8:18
收件人:&nbsp;"weihuayi/fealpy"<fealpy@noreply.github.com&gt;;
抄送:&nbsp;"天官赐福"<1563025275@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [weihuayi/fealpy] 魏老师，我是mac系统，最后验证的时候发现没有安装到conda下，不知道哪里错了 (#18)





 
@yoczhang virtual env 是一个好的手段， 但考虑到大部分人对 Python 还不是很熟悉， 安装过程要尽量的简单，不涉及太多的概念和技巧。 他的这种情况，有点装乱了。

**Comment by yoczhang** [comment, marked as answer]
不用的, Anaconda 就够了, 如果你不怕麻烦的话, [Mac 下 Anaconda 安装 fealpy](https://blog.csdn.net/mifangdebaise/article/details/88778115) 提供了用 pip 安装 fealpy 到 virtual env 的方法.
你需要注意的只是将文中的路径替换成你自己相应的路径下就可以了, 比如 '/anaconda3/envs/myFealpy/bin/' 对应你的路径`/Users/YourName/anaconda3/envs/myFealpy/bin/`.   等等

**Comment by liyingyuan** [comment]
感谢您们的耐心解答🙏，我的问题解决啦，是没有在终端激活base。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "weihuayi/fealpy"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年7月7日(星期二) 晚上8:38
收件人:&nbsp;"weihuayi/fealpy"<fealpy@noreply.github.com&gt;;
抄送:&nbsp;"天官赐福"<1563025275@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [weihuayi/fealpy] 魏老师，我是mac系统，最后验证的时候发现没有安装到conda下，不知道哪里错了 (#18)





 
不用的, Anaconda 就够了, 如果你不怕麻烦的话, Mac 下 Anaconda 安装 fealpy 提供了用 pip 安装 fealpy 到 virtual env 的方法.
 你需要注意的只是将文中的路径替换成你自己相应的路径下就可以了, 比如 '/anaconda3/envs/myFealpy/bin/' 对应你的路径/Users/YourName/anaconda3/envs/myFealpy/bin/.   等等
